proto/cometbft/services/block_results/v1beta1/block_results.proto

```diff
@@ -12,14 +12,22 @@ message GetBlockResultsRequest {
   int64 height = 1;
 }
 
-message GetLatestBlockResultsRequest {
-}
+message GetLatestBlockResultsRequest {}
 
 message GetBlockResultsResponse {
-  int64    height                                                  = 1;
-  repeated cometbft.abci.v1beta3.ExecTxResult txs_results                = 2;
-  repeated cometbft.abci.v1beta2.Event finalize_block_events             = 3;
-  repeated cometbft.abci.v1beta1.ValidatorUpdate validator_updates       = 4;
-  cometbft.types.v1beta3.ConsensusParams         consensus_param_updates = 5;
-  bytes                                    app_hash                = 6;
+  int64 height = 1;
+  repeated cometbft.abci.v1beta3.ExecTxResult txs_results = 2;
+  repeated cometbft.abci.v1beta2.Event finalize_block_events = 3;
+  repeated cometbft.abci.v1beta1.ValidatorUpdate validator_updates = 4;
+  cometbft.types.v1beta3.ConsensusParams consensus_param_updates = 5;
+  bytes app_hash = 6;
+}
+
+message GetLatestBlockResultsResponse {
```

Comment:
So this is now an exact copy of `GetBlockResultsResponse` above.
Should we rather define a common `BlockResults` message and reuse it in both response types?

Reply:
Why not coalescing both methods (`GetBlockResults` and `GetLatestBlockResults`) into one, `GetBlockResults`, and letting `height = 0` denote latest height? This is used in many of our JSON-RPC methods, so users are used to it.

Reply:
This split also exists in the block service, between `GetByHeight` and `GetLatest`.

There are important semantic differences between the two as hinted above:  `GetBlockResults` always returns the same response (in success case) for the same input, while the response of `GetLatestBlockResults` varies with time.
Protobuf has a way to mark down these semantics for binding generators, which we might decide to use at some later point. I can also imagine a proxying service which would cache data for `GetBlockResults`, but query latest node state for `GetLatestBlockResults`.
It would be surprising to have a non-idempotent special case for the otherwise idempotent getter method, and the difference in behavior, without distinction in name, increases potential for inadvertent misuse. I'd argue that it's JSON-RPC that muddles these cases too much, but it's too late to fix.

Reply:
> It would be surprising to have a non-idempotent special case for the otherwise idempotent getter method

Except that's what ADR-101 actually specified and it's even documented in the protos, but then @thanethomson thought it necessary to introduce the "latest" methods in #1209 and as one of the review changes in #1168. I can't find the exact rationale, but I think it's somewhere close to the reasons I pointed out above.

Reply:
I think no one objected to your proposal in Slack, so I'd advice to do the refactoring needed in this PR to follow your proposal (including any change necessary to ADR101).
Your propoal, IIRC, is to remove the "special behaviour of height 0", and keep two methods: one idempotent (and so, cache-able) and the other getting the latest info.

Reply:
I have filed an RFC to discuss the matter  in detail: #1545

Reply:
While that issue waiting its due discussion, the simplest non-intrusive change here is to duplicate the response struct with the same list of fields, so that it is binary compatible with the `tendermint` definition in the main branch.

Reply:
I'd suggest we go ahead and merge this, and tackle the outcome of this discussion in a further PR